boot/seal.go

```diff
@@ -211,24 +275,87 @@ func resealKeyToModeenv(rootdir string, model *asserts.Model, modeenv *Modeenv,
 		bootloader.RoleRunMode:  bl.Name(),
 	}
 
+	authKeyFile := filepath.Join(dirs.SnapFDEDirUnder(rootdir), "tpm-policy-auth-key")
+	if err := resealRunObjectKeys(pbc, authKeyFile, roleToBlName); err != nil {
+		return err
+	}
+	logger.Debugf("resealing (%d) succeeded", nextCount)
+
+	// reseal the fallback object
+	rpbc := toPredictableBootChains(recoveryBootChains)
+
+	var nextFallbackCount int
+	needed, nextFallbackCount, err = isFallbackResealNeeded(rpbc, rootdir, expectReseal)
+	if err != nil {
+		return err
+	}
+	if needed {
+		rpbcJSON, _ := json.Marshal(rpbc)
+		logger.Debugf("resealing (%d) to recovery boot chains: %s", nextCount, rpbcJSON)
+
+		roleToBlName := map[bootloader.Role]string{
+			bootloader.RoleRecovery: rbl.Name(),
+			bootloader.RoleRunMode:  bl.Name(),
+		}
+
+		if err := resealRunObjectKeys(pbc, authKeyFile, roleToBlName); err != nil {
```

Comment:
this looks wrong? shouldn't it say fallback?

Reply:
It's completely wrong. Fixing it.